### DISCUSSION TITLE
Base de CEPs

### DISCUSSION THREAD

**Discussion by rserafim** [BrasilAPI/BrasilAPI#334] [General]
Fala pessoal, já temos a lista de todos os ceps para download? Tenho uma base gigantes pra validar.

**Comment by LorhanSohaky** [comment]
Ainda não temos isso, mas se quiser existe o https://cepaberto.com que fornece estes dados

**Comment by lucianopf** [comment]
@rserafim boa tarde, só pra confirmar vc conseguiu esses dados?

Temos alguém abusando de nossos serviços e estamos em busca pra conversar =)

**Reply by RodriAndreotti** [reply]
@lucianopf imagino que vai chegar uma hora em que um api key será necessário... rs

**Reply by rserafim** [reply]
@lucianopf acabei fazendo um catado de vários, não achei nenhum completo. Mas 95% é desse que vc passou acima.

**Reply by lucianopf** [reply]
Pouts @RodriAndreotti fico tão triste com isso 😢 

É tão mais legal a gnt conseguir prover um serviço full open, gratis e sem nenhuma autenticação 😭

**Reply by RodriAndreotti** [reply]
Eu concordo plenamente contigo... e ai caímos naquela história: Quando um não sabe usar todos pagam.
A história da autenticação seria mais para tentar impedir abusos mesmo.
Por que, se não tiver uma cota, vai chegar uma hora que os servers da Vercel não darão mais conta.

**Reply by LorhanSohaky** [reply]
Um simples rate limit não resolve?

**Reply by RodriAndreotti** [reply]
Talvez um throttle tbm fosse interessante.
Mas como vamos identificar quem está acessando? Por IP é meio complicado, porque 90% dos ip's do Brasil são dinâmicos :-/

**Reply by murilohns** [reply]
@RodriAndreotti, autenticação não resolve por enquanto, pq vamos continuar tomando pancada na API, mesmo com autenticação

@LorhanSohaky, rate limit não resolve também pq o cara tá acessando de vários IP's diferentes, e a quantidade de request por IP não é alta, mas são muitos IP's. Não dá pra bloquear os IP's também pq parecem estar ser de infra compartilhada, daí podemos afetar outras pessoas

**Reply by RodriAndreotti** [reply]
Realmente com a autenticação ainda receberíamos o request.

Será que isso está vindo de pod's de k8s então, algum crawler talvez?

**Comment by diego91964** [comment]
Estava pensando em uma arquitetura que consiga garantir que o projeto
continue livre, gratuito e ainda aumente a disponibilidade.

 Fiz um rascunho aqui para melhorar a explicação, o Brasil API é a
aplicação conforme é hoje.

![rascunho](https://user-images.githubusercontent.com/887760/161332673-db294f7d-4eb8-4cda-985a-5581448f289d.png)



A ideia é criarmos um Load Balancer com as seguintes atribuições: controle
de cache e controle de uma carteira de tokens.

Quando alguém deseja se tornar um consumidor do BrasilAPI ele teria duas
opções, a primeira é como é hoje, porém com um delay que garanta a saúde do
serviço (Gerenciado pelo Load Balancer).

A segunda opção é especificar um nó que seja um ‘provider’ de cache que se
conecta ao loadbalancer, o provider aqui descrito ajudaria a liberar a
carga de trabalho do nó principal e ao mesmo tempo, de acordo que fica
disponível no fornecimento de serviço, ele vai gerando tokens em uma
carteira para que possam ser consumidos por consumidores que utilizem o
código da carteira.

A ideia da proposta é garantir que quanto mais consumidores o BrasilAPI
tenha mais nós distribuídos de providers ela tenha.


Em sex., 1 de abr. de 2022 às 09:30, Rodrigo Teixeira Andreotti <
***@***.***> escreveu:

> Realmente com a autenticação ainda receberíamos o request.
>
> Será que isso está vindo de pod's de k8s então, algum crawler talvez?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/BrasilAPI/BrasilAPI/discussions/334#discussioncomment-2485538>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAGYXUGE5GPX7VTX7OR3VKTVC3T6JANCNFSM5BM3FAYA>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>


-- 
-------------------------------------
Diego A.Silva.

Jug Leader UaiJUG

@diego91964

**Reply by murilohns** [reply]
@diego91964 a imagem não carregou 😬

**Reply by RodriAndreotti** [reply]
Voltei agora para ler melhor, achei interessante a ideia, mas fiquei curioso com a imagem... rs

**Reply by diego91964** [reply]
![rascunho](https://user-images.githubusercontent.com/887760/161332804-f0832d97-dc39-447c-8bcf-026ef7784d14.png)

No caso precisaríamos construir um padrão de provider que o responsável poderia subir em um nó e fornecer o cache distribuído.

Segue a imagem em anexo.

**Reply by RodriAndreotti** [reply]
Achei a ideia interessante e promissora... rs

**Reply by rserafim** [reply]
Sou noob, mas ja que esta na Vercel, usar edge functions nao funcionaria? https://vercel.com/features/edge-functions

**Reply by LorhanSohaky** [reply]
@rserafim , parece interessante. @lucianopf @murilohns deem uma olhada nisso

**Reply by lucianopf** [reply]
> Sou noob, mas ja que esta na Vercel, usar edge functions nao funcionaria? https://vercel.com/features/edge-functions

A treta aqui é a quantidade de requests que recebemos vs o quanto podemos usar na Vercel =/

Recebemos em média 65 milhões de request no mês e na Vercel só teriamos grauidade do primeiro milhão 😭 

E sobre a proposta eu to aqui tentando refletir ainda pq tá me parecendo que a gnt pretende criar uma infra bem mais complexa com custos envolvidos num projeto que tem zero orçamento 🤔
